ibm/service/database/validators.go

```diff
@@ -0,0 +1,98 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package database
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/IBM/cloud-databases-go-sdk/clouddatabasesv5"
+)
+
+/* TODO move other validators in here */
+
+/* VERSION VALIDATOR */
```

Comment:
Similar to [`DatabaseUser`](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/database/resource_ibm_database.go#L63-L68) there can be an internal `Version` struct that would be easier to mock/test than the SDK Client

```go
type Version struct {
  Version string,
  Status string,
  IsPreferred bool,
  Transitions []VersionTransition
}

type VersionTransition struct {
  Method string,
  FromVersion string,
  ToVersion string,
}
```

This can be mocked and have methods defined on the struct that can be tested i.e. `version.CanUpgradeTo(newVersion) => true`

Database User tests: https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/database/resource_ibm_database_test.go

Reply:
Updated